ECOPROJECT-12: Documentation for the Poison Pill Operator (#35974)
openshift/openshift-docs: modules/eco-poison-pill-operator-troubleshooting.adoc

// Module included in the following assemblies:
//
// * nodes/nodes/eco-poison-pill-operator.adoc

[id="troubleshooting-poison-pill-operator_{context}"]
= Troubleshooting the Poison Pill Operator

[id="general-troubleshooting-poison-pill-operator_{context}"]
== General troubleshooting 

Issue::
You want to troubleshoot issues with the Poison Pill Operator.

Resolution::
Check the Operator logs.

[id="checking-daemon-set_{context}"]
== Checking the daemon set 
Issue:: The Poison Pill Operator is installed but the daemon set is not available.

Resolution:: Check the Operator logs for errors or warnings.

[id="unsuccessful_remediation{context}"]
== Unsuccessful remediation 
Issue:: An unhealthy node was not remediated.

> Resolution:: Check and verify if the `PoisonPillRemediation` CR is created. Run the following command:

Reviewer: Consider a small revision for clarity:
```suggestion
Resolution:: Verify that the `PoisonPillRemediation` CR was created by running the following command:
```

+
[source,terminal]
----
$ oc get ppr -A
----
+
If the `MachineHealthCheck` controller did not create the `PoisonPillRemediation` CR when the node turned unhealthy, check the logs of the `MachineHealthCheck` controller. Additionally, ensure that the `MachineHealthCheck` CR includes the required specification to use the remediation template.
+

> If the `PoisonPillRemediation` CR is created, ensure that its name matches the unhealthy node or the machine object.

Reviewer: s/is created/was created/ (because it already happened)

// Module included in the following assemblies:
//
// * nodes/nodes/eco-poison-pill-operator.adoc

[id="troubleshooting-poison-pill-operator_{context}"]
= Troubleshooting the Poison Pill Operator

> 

Reviewer: Please remove the extra line break

[id="general-troubleshooting-poison-pill-operator_{context}"]

> == General Troubleshooting 

Reviewer: This should be sentence case ("General troubleshooting")

Issue::
You want to troubleshoot issues with the Poison Pill Operator.

Resolution::
Check the Operator logs.

[id="checking-daemon-set_{context}"]
== Checking the daemon set 
Issue:: The Poison Pill Operator is installed but the daemon set is not available.

Resolution:: Check the Operator logs for errors or warnings.

[id="unsuccessful_remediation{context}"]
== Unsuccessful remediation 
Issue:: An unhealthy node was not remediated.

Resolution:: Check and verify if the `PoisonPillRemediation` CR is created. Run the following command:
+
[source,terminal]
----

> $ oc get namespace <machinehealthcheck>

Reviewer 2: should be `oc get ppr -A`

----
+

> If the `MachineHealthCheck` controller did not create the `PoisonPillRemediation` CR when the node turned unhealthy, check the logs of the `MachineHealthCheck` controller.

Reviewer 3: maybe it is worth mentioning that a PoisonPillRemediationTemplate resource needs to exists (to be created manually)
in the cluster, for the machine health check to create the ppr

+
If the `PoisonPillRemediation` CR is created, ensure that its name matches the unhealthy node or the machine object.

[id="daemon-set-exists_{context}"]
== Daemon set exists even after uninstalling the Poison Pill Operator 
Issue:: The Poison Pill daemon set exists even after after uninstalling the Operator.

Resolution:: To remove the Poison Pill daemon set, manually delete the `PoisonPillConfig` CR. Run the following command:  
+
[source,terminal]
----

> $ oc delete pods <poison-pill-daemonset> -n <namespace>

Reviewer 2: should be `oc delete ds ...`

// Module included in the following assemblies:
//
// * /machine_management/eco-poison-pill-operator.adoc

[id="troubleshooting-poison-pill-operator_{context}"]
= Troubleshooting the Poison Pill Operator

[id="general-troubleshooting-poison-pill-operator_{context}"]
== General Troubleshooting 

Issue::
You want to troubleshoot issues with the Poison Pill Operator.

Resolution::

> Enable the debug flag in the `poison-pill` object. The `debug` flag increases

Reviewer 4: The example CR in "Understanding the Poison Pill Operator configuration" does not list a "debug" flag.

Author: Updated the resolution section to check pod logs.

the verbosity of the daemons that run in the `DaemonSet` pods and run the AIDE
checks.

[id="checking-daemon-set_{context}"]
== Checking the daemon set 
Issue:: The Poison Pill Operator is installed but the daemon set is not available.

Resolution:: Check the Operator logs for errors or warnings.

[id="unsuccessful_remediation{context}"]
== Unsuccessful remediation 
Issue:: An unhealthy node was not remediated.

Resolution:: Check and verify if the `PoisonPillRemediation` CR is created. To do so, run the following command:
+
[source,bash,options="nowrap",role="white-space-pre"]
----
oc get namespace <machinehealthcheck>
----

> If the `PoisonPillRemediation` CR is not created when the node turned unhealthy, check the logs of the `MachineHealthCheck` controller.

Reviewer 5: General comment: what is creating the `PoisonPillRemediation` CR? If we can answer that, we can address the passive voice.

Reviewer 4: The `MachineHealthCheck` controller

If the `PoisonPillRemediation` CR is created, ensure that its name matches the unhealthy node or the machine object.

[id="daemon-set-exists_{context}"]
== Daemon set exists even after uninstalling the Poison Pill Operator 
Issue:: The Poison Pill daemon set exists even after after uninstalling the Operator.

> Resolution:: To remove the Poison Pill daemon set, manually delete the `PoisonPillConfig` CR.

Reviewer 6: For troubleshooting 'Unsuccessful remediation' you provide the command 'To do so, run the following command:' 
I think it would also be helpful to include the command for troubleshooting 'Daemon set exists even after uninstalling the Poison Pill Operator'.

Author: Thank you; added the command
